src/core/apply.js

```diff
@@ -52,12 +54,24 @@ const callback = (obj, prop) => {
  * @since 0.4.0
  */
 const apply = (obj, path, operation) => {
-  const walkPath = (curObj, curPath) => {
+  const walkPath = (curObj, curPath, doCopy = true) => {
     const [prop, ...pathRest] = curPath
 
-    const value = callback(curObj, prop)
+    if (isSlice(prop)) {
+      const [start, end] = getSliceBounds(prop, length(curObj))
+
+      const newArr = copy(curObj, true)
+
+      for (let i = start; i < end; i++)
+        walkPath(newArr, [i, ...pathRest], false)
+
+      return newArr
+    }
+
+    const value = isNil(curObj) ? undefined : curObj[prop]
 
-    const newObj = copy(curObj, isArrayProp(prop))
+    let newObj = curObj
+    if (doCopy) newObj = copy(curObj, isIndex(prop))
```

Comment:
Why do we need it ?

Reply:
When we are iterating on a slice, only one copy is necessary, so this is to avoid several copies